File: Android/TagMaster/src/depollsoft/tagmaster/MeActivity.java
package depollsoft.tagmaster;

import android.app.Activity;
import android.content.Intent;
import android.os.Bundle;
import android.view.Menu;
import android.view.MenuItem;

import com.bindroid.converters.AdapterConverter;
import com.bindroid.converters.BoolConverter;
import com.bindroid.trackable.TrackableCollection;
import com.bindroid.ui.UiBinder;
import com.flurry.android.FlurryAgent;
import com.parse.ParseException;
import com.parse.ParseObject;
import com.parse.ParseUser;
import com.parse.RefreshCallback;

import depollsoft.lib.compat.ui.ActionBars;
import depollsoft.lib.compat.ui.MenuItems;
import depollsoft.lib.ui.ChangelogViewer;

public class MeActivity extends Activity {
  public MeActivity() {
  }

  public TrackableCollection<Integer> getFavoriteIds() {
    return FavoritesModel.getFavoriteIds();
  }

  public boolean getHasActionBar() {
    return ActionBars.hasActionBar(this);
  }

  @Override
  protected void onCreate(Bundle savedInstanceState) {
    super.onCreate(savedInstanceState);
    this.setContentView(R.layout.meview);

    UiBinder.bind(this, R.id.favoritesItemsControl, "Adapter", "FavoriteIds", new AdapterConverter(
        FavoriteTagItemView.class, false, true));

    UiBinder.bind(this, R.id.titleLayout, "Visibility", "HasActionBar", new BoolConverter(true));

    ActionBars.setCustomTitle(this, R.layout.titleview);

    if (ParseUser.getCurrentUser() != null) {
      try {
        ParseUser.getCurrentUser().refreshInBackground(new RefreshCallback() {

          @Override
          public void done(ParseObject obj, ParseException err) {
            if (err != null) {
              return;
            }
            FavoritesModel.restoreFromUser();
            TeachableTagsModel.restoreFromUser();
          }
        });
      }
      catch (Exception e) {
      }
    }

    ChangelogViewer viewer = new ChangelogViewer(this, this.getString(R.string.Changelog));
    viewer.setTitle("Tag Master Changelog");
    viewer.setIcon(this.getResources().getDrawable(R.drawable.icon));
    viewer.showIfAppropriate();
  }

  @Override
  public boolean onCreateOptionsMenu(Menu menu) {
    this.getMenuInflater().inflate(R.menu.memenu, menu);

    MenuItems
        .setShowAsAction(menu.findItem(R.id.settingsMenuItem), MenuItems.SHOW_AS_ACTION_ALWAYS);
    return true;
  }

  @Override
  protected void onDestroy() {
    super.onDestroy();
  }

  @Override
  public boolean onOptionsItemSelected(MenuItem item) {
    if (item.getItemId() == R.id.settingsMenuItem) {
      Intent i = new Intent(this, SettingsActivity.class);
      this.startActivity(i);
      return true;
    }
    return super.onOptionsItemSelected(item);
  }

  @Override
  protected void onResume() {
    super.onResume();
  }

  @Override
  public boolean onSearchRequested() {
    Intent i = new Intent(this, TagSearchActivity.class);
    this.startActivity(i);
    return true;
  }

  @Override
  protected void onStart() {
    super.onStart();
    FlurryAgent.onStartSession(this, "V5L1948BNDQCKZFPARJ9");
  }

  @Override
  protected void onStop() {
    super.onStop();
    FlurryAgent.onEndSession(this);
  }

}

File: Android/TagMaster/src/depollsoft/tagmaster/MeActivity.kt
package depollsoft.tagmaster

import android.content.Intent
import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import android.view.Menu
import android.view.MenuItem
import android.view.View
import com.bindroid.converters.AdapterConverter
import com.bindroid.trackable.TrackableCollection
import com.bindroid.ui.UiBinder
import com.parse.ParseUser
import com.parse.RefreshCallback
import depollsoft.lib.compat.ui.MenuItems
import depollsoft.lib.ui.ChangelogViewer

class MeActivity : AppCompatActivity() {

    val favoriteIds: TrackableCollection<Int>
        get() = FavoritesModel.getFavoriteIds()

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        this.setContentView(R.layout.meview)

        UiBinder.bind(this, R.id.favoritesItemsControl, "Adapter", "FavoriteIds", AdapterConverter(
                FavoriteTagItemView::class.java, false, true))

        this.supportActionBar?.title = "Tag Master".makeTitleString(this)

        if (ParseUser.getCurrentUser() != null) {
            try {
                ParseUser.getCurrentUser().refreshInBackground(RefreshCallback { _, err ->
                    if (err != null) {
                        return@RefreshCallback
                    }
                    FavoritesModel.restoreFromUser()
                    TeachableTagsModel.restoreFromUser()
                })
            } catch (e: Exception) {
            }

        }

        val viewer = ChangelogViewer(this, this.getString(R.string.Changelog))
        viewer.setTitle("Tag Master Changelog")
        viewer.setIcon(R.mipmap.ic_launcher)
        viewer.showIfAppropriate()

        findViewById<View>(R.id.searchButton).setOnClickListener {
            val i = Intent(this, TagSearchActivity::class.java)
            startActivity(i)
        }
    }

    override fun onCreateOptionsMenu(menu: Menu): Boolean {
        this.menuInflater.inflate(R.menu.memenu, menu)

        MenuItems
                .setShowAsAction(menu.findItem(R.id.settingsMenuItem), MenuItems.SHOW_AS_ACTION_ALWAYS)
        return true
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        if (item.itemId == R.id.settingsMenuItem) {
            val i = Intent(this, SettingsActivity::class.java)
            this.startActivity(i)
            return true
        }
        return super.onOptionsItemSelected(item)
    }

    override fun onSearchRequested(): Boolean {
        val i = Intent(this, TagSearchActivity::class.java)
        this.startActivity(i)
        return true
    }
}
